geonode/geoserver/manager.py

```diff
@@ -400,114 +402,122 @@ def remove_permissions(self, uuid: str, /, instance: ResourceBase = None) -> boo
+
+                                    if not created:
+                                        purge_geofence_dataset_rules(_resource.get_self_resource())
+
+                                    # Owner & Managers
+                                    perms = [
+                                        "view_resourcebase",
+                                        "change_dataset_data",
+                                        "change_dataset_style",
+                                        "change_resourcebase",
+                                        "change_resourcebase_permissions",
+                                        "download_resourcebase"]
+                                    sync_geofence_with_guardian(_resource, perms, user=_owner)
+                                    gf_services = _get_gf_services(_resource, perms)
+                                    _, _, _disable_dataset_cache, _, _, _ = get_user_geolimits(_resource, _owner, None, gf_services)
                                     _disable_cache.append(_disable_dataset_cache)
 
-                            # Anonymous
-                            if anonymous_can_view:
-                                sync_geofence_with_guardian(instance, VIEW_PERMISSIONS, user=None, group=None)
-                                gf_services = _get_gf_services(instance, VIEW_PERMISSIONS)
-                                _, _, _disable_dataset_cache, _, _, _ = get_user_geolimits(instance, None, None, gf_services)
-                                _disable_cache.append(_disable_dataset_cache)
-
-                            if anonymous_can_download:
-                                sync_geofence_with_guardian(instance, DOWNLOAD_PERMISSIONS, user=None, group=None)
-                                gf_services = _get_gf_services(instance, DOWNLOAD_PERMISSIONS)
-                                _, _, _disable_dataset_cache, _, _, _ = get_user_geolimits(instance, None, None, gf_services)
-                                _disable_cache.append(_disable_dataset_cache)
-
-                        if _disable_cache:
-                            filters, formats = _get_gwc_filters_and_formats(_disable_cache)
-                            try:
-                                _dataset_workspace = get_dataset_workspace(instance.get_real_instance())
-                                toggle_dataset_cache(f'{_dataset_workspace}:{instance.get_real_instance().name}', filters=filters, formats=formats)
-                            except Dataset.DoesNotExist:
-                                pass
-                    else:
-                        instance.set_dirty_state()
+                                    _resource_groups, _group_managers = _resource.get_group_managers(group=_resource.group)
+                                    for _group_manager in _group_managers:
+                                        sync_geofence_with_guardian(_resource, perms, user=_group_manager)
```

Comment:
Can we wrap those 4 methods in a single one called `self._sync_permissions_and_disable_cache` and let it internally do these things?
The method is really long, at least we can improve the readability

Reply:
Done